### PR TITLE
[macos] Stop building classic and use pre-built binaries

### PR DESCRIPTION
**Pull request by chamons**
- Current 4.0.0.214 (15.5 P4) but will be updated to 15.5 final

### CODE REVIEW

**Comment by chamons** [comment]
Testing removal of classic from src builds as first step on PMCS removal.

The :( part will be the checked in binaries:

```
-rwxr-xr-x    1 donblas  staff    35K Nov 16 16:14 XamMac.CFNetwork.dll
-rw-r--r--    1 donblas  staff    10K Nov 16 16:14 XamMac.CFNetwork.dll.mdb
-rwxr-xr-x    1 donblas  staff    14M Nov 16 16:14 XamMac.dll
-rw-r--r--    1 donblas  staff   4.6M Nov 16 16:14 XamMac.dll.mdb
```

but I'm not sure of a way around that.

There is a maccore change to go along w\ this. That would have to go in first (and change mk/xamarin.mk back to master).

File: src/Makefile [hunk 1]
```diff
@@ -594,17 +591,7 @@ $(MONO_PATH)/mcs/class/lib/xammac/reference_Facades/netstandard.dll: $(MAC_BUILD
 $(MAC_BUILD_DIR)/mobile/Facades/netstandard.dll: $(MONO_PATH)/mcs/class/lib/xammac/reference_Facades/netstandard.dll $(MAC_BUILD_DIR)/mobile/Facades
 	$(Q) cp $< $@
 
-$(MAC_BUILD_DIR)/compat/XamMac.CFNetwork.dll: $(MAC_CFNETWORK_SOURCES) $(MAC_BUILD_DIR)/compat/XamMac.dll $(SN_KEY)
```
Comment (rolfbjarne) [inline src/Makefile]: If you do something like this:

```
$(MAC_BUILD_DIR)/compat/XamMac.CFNetwork.dll: $(MACIOS_BINARIES_PATH)/XamMac.CFNetwork.dll | $(MAC_BUILD_DIR)/compat
	$(Q) cp $< $@
	$(Q) cp $<.mdb $@.mdb

$(MAC_BUILD_DIR)/compat/XamMac.dll: $(MACIOS_BINARIES_PATH)/XamMac.dll | $(MAC_BUILD_DIR)/compat
	$(Q) cp $< $@
	$(Q) cp $<.mdb $@.mdb
```

you don't need the changes below nor the changes in maccore anymore.

**Comment by chamons** [comment]
@spouliot I'm not completely seeing why the classic binaries should live in another repro. If we make it a submodule then won't the initial sync pull it in anyway (so we aren't saving disk/network bandwidth.

If we were planning on updating it more than a handful of times at most I could see it more.

I'll make a new macios-binaries if you want though.

File: src/Makefile (hunk 1 above)
Comment (chamons) [reply src/Makefile]: Good idea, let me try that.

**Comment by rolfbjarne** [comment]
@chamons once in a repo, the binaries will never go away. This means that once we kill classic, we'll still have to download the binaries (because they're in the history), even if all traces of them have vanished from all branches we care about.

Using a different repo will make sure we don't pay that price once we don't need the binaries anymore.

**Comment by chamons** [comment]
Once macios-binaries is created I'll fix up this PR and close the maccore one.

**Comment by chamons** [comment]
Latest changes build fine locally and remove the checked in binary. I did not run tests locally, but the bots should run everything tonight.

File: .gitmodules [hunk 2]
```diff
@@ -29,3 +29,6 @@
     path = external/Xamarin.MacDev
     url = ../../xamarin/Xamarin.MacDev
     branch = master
+[submodule "external/macios-binaries"]
+	path = external/macios-binaries
+	url = https://github.com/xamarin/macios-binaries
```
Comment (rolfbjarne) [inline .gitmodules]: Use a relative url (look at the others) instead.

Also indentation is different than the others (whitespace issue)?

Comment (chamons) [reply .gitmodules]: Strange, it is what git submodule add did to the file. I did not manually change it.

Let me fix this up.

**Comment by chamons** [comment]
Test failures are just device tests that are skipped due to me adding run-all-tests tag.

**Pull request merged**